tests/src/test/scala/com/nvidia/spark/rapids/CastOpSuite.scala

```diff
@@ -425,6 +383,14 @@ class CastOpSuite extends GpuExpressionTestSuite {
       col("c1").cast(FloatType))
   }
 
+  ignore("overflow double strings") {
+    testSparkResultsAreEqual("overflow double strings", badDoubleStringsDf,
```

Comment:
`badDoubleStringsDf` is not as comprehensive as I would like it to be. The values in it are big enough to force an overflow, but there are still a number of corner cases where according to https://github.com/rapidsai/cudf/issues/5225 are not being covered. The patch to fix it https://github.com/rapidsai/cudf/pull/7410 is not in and not working 100% either in these same overflow cases. I am happy to have these go in as it, but I would like to see a follow on issue to cover the corner cases that are missing, and have a comment in `badDoubleStringsDf` pointing to it.  Would be good to look at the float one too and make sure it is also not suffering from the same issue.

Reply:
Hi @revans2, I tried with many other similar (slightly overflow) cases. And I found GPU can produce correct results for almost all of them, except some special cases included in `badDoubleStringsDf`.
For float type, I failed to find any mismatching case.
For double type, I only found numbers within range [1.79769313486231581E308, 1.797693134862316E308) produce inconsistent results with spark.